### PR TITLE
Add new TASH Command named stkopt

### PR DESCRIPTION
**Pull request by alswnzla**

### CODE REVIEW

File: apps/system/utils/kdbg_stackopt.c [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+				break;
+			}
+		}
+#endif
+		pid = task_create(args[0], pri, stack_size, cb , &args[1]);
+	}
+	return pid;
+}
+
+int get_frame_size(int argc, char **args)
```
Comment (jeongchanKim) [inline apps/system/utils/kdbg_stackopt.c]: This function is only used in this file. so let's change static

File: apps/system/utils/kdbg_stackopt.c [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+	while (args[args_idx]) {
+		strtablen += (strlen(args[args_idx]) + 1);
+		args_idx++;
+	}
+	
+	frame_size = STACK_ALIGN_UP(argvlen + strtablen);
+	return frame_size;
+}
+
+void print_measure_result(pid_t pid, int argc, char **args)
```
Comment (jeongchanKim) [inline apps/system/utils/kdbg_stackopt.c]: same as above

File: apps/system/utils/kdbg_stackopt.c [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+		printf("Cannot find APP\n");
+		goto usage;
+	} else {
+		waitpid(pid, &status, 0);
+		print_measure_result((pid_t)pid, argc - 1, &args[1]);
+	}
+	return OK;
+
+usage:
+	printf("\nUsage: stkprof NAME [ARGS]\n");
```
Comment (jeongchanKim) [inline apps/system/utils/kdbg_stackopt.c]: change "stkprof" to "stkopt"
do you have any special reason that use [ ] only for ARGS?

File: apps/system/utils/kdbg_stackopt.c [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+ */
+
+#ifdef __ARM_EABI__
+#define CONFIG_STACK_ALIGNMENT 8
+#else
+#define CONFIG_STACK_ALIGNMENT 4
+#endif
+#endif
+
+#define STACK_ALIGN_MASK    (CONFIG_STACK_ALIGNMENT-1)
```
Comment (jeongchanKim) [inline apps/system/utils/kdbg_stackopt.c]: use space near operator

File: apps/system/utils/kdbg_stackopt.c [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+
+bool find_app(TASH_CMD_CALLBACK *cb, char *name)
+{
+	int task_idx;
+	bool flag = false;
+	char str[CONFIG_TASK_NAME_SIZE + 1];
+	int count = tash_get_cmdscount();
+	
+	for (task_idx = 0; task_idx < count; task_idx++) {
+		tash_get_cmdpair(str, cb, task_idx);
```
Comment (sunghan-chang) [inline apps/system/utils/kdbg_stackopt.c]: Don't you need to check return value? If it doesn't, add (void).

File: apps/system/utils/kdbg_stackopt.c [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+extern tash_taskinfo_t tash_taskinfo_list[];
+
+bool find_app(TASH_CMD_CALLBACK *cb, char *name)
+{
+	int task_idx;
+	bool flag = false;
+	char str[CONFIG_TASK_NAME_SIZE + 1];
+	int count = tash_get_cmdscount();
+	
+	for (task_idx = 0; task_idx < count; task_idx++) {
```
Comment (sunghan-chang) [inline apps/system/utils/kdbg_stackopt.c]: Why name is task_idx? Isn't it cmd_idx?

File: apps/system/utils/kdbg_stackopt.c [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+#define STACK_ALIGN_MASK    (CONFIG_STACK_ALIGNMENT-1)
+#define STACK_ALIGN_UP(a)   (((a) + STACK_ALIGN_MASK) & ~STACK_ALIGN_MASK)
+
+/****************************************************************************
+ * Extern Data
+ ****************************************************************************/
+extern struct stkmon_save_s stkmon_arr[CONFIG_MAX_TASKS * 2];
+extern tash_taskinfo_t tash_taskinfo_list[];
+
+bool find_app(TASH_CMD_CALLBACK *cb, char *name)
```
Comment (sunghan-chang) [inline apps/system/utils/kdbg_stackopt.c]: add static

File: apps/system/utils/kdbg_stackopt.c (hunk 3 above)
Comment (sunghan-chang) [inline apps/system/utils/kdbg_stackopt.c]: Use args[0] instead of fixed string.

Comment (sunghan-chang) [inline apps/system/utils/kdbg_stackopt.c]: And you should notice what NAME is and [ARGS] is.

File: apps/system/utils/kdbg_stackopt.c [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+			break;
+		}
+	}
+
+	printf("-------------------------------------------------------\n");
+	printf(">> Stack Size Measure Finish\n");
+	printf("APP Name : %s\n", stkmon_arr[stkmon_idx].chk_name);
+	printf("Allocate %d stack size for measure\n",  stkmon_arr[stkmon_idx].chk_stksize);
+	printf("Optimal Stack Size is %d\n", stkmon_arr[stkmon_idx].chk_peaksize + get_frame_size(argc, args));
+	printf("If the arguments change, optimal stack size will change\n");
```
Comment (sunghan-chang) [inline apps/system/utils/kdbg_stackopt.c]: can is better than will.
```
printf("If the arguments are changed, optimal stack size can be changed\n");
```

File: apps/system/utils/kernelcmd.c [hunk 9]
```diff
@@ -58,6 +58,9 @@ const static tash_cmdlist_t kdbg_cmds[] = {
 #if defined(CONFIG_ENABLE_STACKMONITOR)
 	{"stkmon",   kdbg_stackmonitor, TASH_EXECMD_SYNC},
 #endif
+#if defined(CONFIG_ENABLE_STACKOPT)
+	{"stkopt",   kdbg_stackopt   , TASH_EXECMD_ASYNC},
```
Comment (sunghan-chang) [inline apps/system/utils/kernelcmd.c]:
```
kdbg_stackopt,       TASH_XXXX
```

**Pull request merged**